Services/Curl.php

```diff
@@ -85,19 +85,23 @@ public function sendRequest($url, $method, array $options = array(), $payload =
 
         $this->curlOptionsHandler->setOptions($options);
         $this->curlOptionsHandler->setOption(CURLOPT_RETURNTRANSFER, true);
+        $this->curlOptionsHandler->setOption(CURLOPT_HEADER, true);
 
         $this->setUrl($url);
         $this->setMethod($method);
         $this->setPayload($payload);
         curl_setopt_array($this->curl, $this->curlOptionsHandler->getOptions());
 
         $curlResponse = $this->execute();
+        $headerSize = curl_getinfo($this->curl, CURLINFO_HEADER_SIZE);
```

Comment:
Let's make a new class "ResponseHeaders" with a ```public static function create($curlResponse, $headerSize)```.

The create function returns an array containing valid headers (the result of http_parse_headers).
So this is what the code would look like here:

```
$curlResponse = $this->execute();
$curlMetaData = (object) curl_getinfo($this->curl);
$headers     = ResponseHeaders::create($curlResponse, $curlMetaData->CURLINFO_HEADER_SIZE);

$this->curlOptionsHandler->reset();
          function_exists('curl_reset') ? curl_reset($this->curl) : curl_setopt_array($this->curl, $this->curlOptionsHandler->getOptions());

return $this->createResponse($content, $headers, $curlMetaData);
```